src/bench_tools.py

```diff
@@ -125,3 +125,12 @@ def radosgw_user_create(self, user, subuser, secret):
                 _cmd, stderr=subprocess.PIPE).decode("UTF-8"))
 
         return _output
+
+
+def fill_zero(device: str):
+    cmd = ["dd", "if=/dev/zero", f"of={device}", "bs=4M", "oflag=sync"]
```

Comment:
I get the need to make sure that the writes are committed to stable storage, but using "sync" seems a bit excessive, imo. You could essentially omit the flag and then call `os.sync` and have the same effects, while making the wipe much faster.

Reply:
> You could essentially omit the flag and then call os.sync and have the same effects, while making the wipe much faster.

Is it gonna be faster actually? It's already with 4M block so sync will be issued each 4M only. If it can be done with one step, I don't see a benefit of separating the step further.

Reply:
`oflag=sync` doesn't just synchronize to stable storage - it bypasses the page cache completely, which is what makes the operation slower in comparison to a simpler accumulation of dirty pages and a flush at the end. I don't have any benchmarks handy to back my claim, though, but I'm willing to bet it's slower, yes.

Reply:
[prep]
```bash
modprobe brd rd_nr=1 rd_size=$((1024*1024))

ls -l /dev/ram0 
brw-rw---- 1 root disk 1, 0 Nov 20 20:37 /dev/ram0

mkdir /sys/fs/cgroup/test_dd

echo $$ | sudo tee /sys/fs/cgroup/test_dd/cgroup.procs

echo "1:0 rbps=$((400*1000*1000)) wbps=$((400*1000*1000))" | sudo tee /sys/fs/cgroup/test_dd/io.max
```

[no flag]

```
# time (dd if=/dev/zero of=/dev/ram0 bs=4M; time sync)
dd: error writing '/dev/ram0': No space left on device
257+0 records in
256+0 records out
1073741824 bytes (1.1 GB, 1.0 GiB) copied, 3.4157 s, 314 MB/s

real	0m0.006s
user	0m0.002s
sys	0m0.001s

real	0m3.424s
user	0m0.002s
sys	0m1.414s
```

[conv=fsync]

```
# time (dd if=/dev/zero of=/dev/ram0 bs=4M conv=fsync; time sync)
dd: error writing '/dev/ram0': No space left on device
257+0 records in
256+0 records out
1073741824 bytes (1.1 GB, 1.0 GiB) copied, 3.38988 s, 317 MB/s

real	0m0.005s
user	0m0.002s
sys	0m0.000s

real	0m3.398s
user	0m0.002s
sys	0m1.372s
```

[oflag=sync]

```
# time (dd if=/dev/zero of=/dev/ram0 bs=4M oflag=sync; time sync)
dd: error writing '/dev/ram0': No space left on device
257+0 records in
256+0 records out
1073741824 bytes (1.1 GB, 1.0 GiB) copied, 2.82269 s, 380 MB/s

real	0m0.007s
user	0m0.003s
sys	0m0.000s

real	0m2.833s
user	0m0.004s
sys	0m1.594s
```

[oflag=direct]

```
# time (dd if=/dev/zero of=/dev/ram0 bs=4M oflag=direct; time sync)
dd: error writing '/dev/ram0': No space left on device
257+0 records in
256+0 records out
1073741824 bytes (1.1 GB, 1.0 GiB) copied, 2.62779 s, 409 MB/s

real	0m0.008s
user	0m0.003s
sys	0m0.000s

real	0m2.640s
user	0m0.008s
sys	0m0.422s
```

[oflag=sync,nocache]
```
# time (dd if=/dev/zero of=/dev/ram0 bs=4M oflag=sync,nocache; time sync)
dd: error writing '/dev/ram0': No space left on device
257+0 records in
256+0 records out
1073741824 bytes (1.1 GB, 1.0 GiB) copied, 2.6623 s, 403 MB/s

real	0m0.010s
user	0m0.002s
sys	0m0.001s

real	0m2.677s
user	0m0.007s
sys	0m1.594s
```